### DISCUSSION TITLE
[v3] Pass arguments to `DeleteAction` in Livewire component

### DISCUSSION THREAD

**Discussion by sprtk-ches** [filamentphp/filament#9076] [Help]
I want to add a `DeleteAction` to my Livewire component but I'm not sure how to pass arguments to it so that I can set the required `record`.

Here's a simplified example

> template
```html
@foreach($messages as $message)
    {{ ($this->deleteMessage)(['message_id' => $message->id])}}
@endforeach
```

> component
```php
public function deleteMessage(): Action
{
    return DeleteAction::make('deleteMessage')
        ->record(fn (array $arguments) => IssueMessage::find($arguments['message_id']));
}
```

This is the closest I got to something that _should_ work. I've also tried setting the record on the `action()` call but that doesn't work.

Is something like that possible? Can I create a "dynamic" `DeleteAction`?

**Comment by sinnbeck** [comment]
Dont you just need it in the action method or?

```php
public function deleteMessage(): Action
{
    return Action::make('deleteMessage')
        ->requiresConfirmation()
        ->label('Delete')
        ->color('danger')
        ->action(fn (array $arguments) => IssueMessage::find($arguments['message_id'])?->delete());
}
```
Or is it because you need it for a modal or something ?

**Reply by sprtk-ches** [reply]
I can use a simple `Action` but I'm losing of the advantages of the `DeleteAction` class, like styling, notifications, the fact that it supports soft-deletes (button gets disabled if the record is soft-deleted) etc.

I can duplicate that functionality but I'm wondering if there's a better way to do it since all that's really stopping me is that `DeleteActions` needs a record and I can't pass one "dynamically".

**Reply by sinnbeck** [reply]
Ah ok. Gave it a try and it seems that on rerender it does not pass the arguments to `->record()` which breaks everything. I might fiddle around with it some more and see if I can find a solution, but currently it does not seem likely.
